### DISCUSSION TITLE
Android CA cert issue

### DISCUSSION THREAD

**Discussion by dbrody2004** [strongswan/strongswan#579] [Q&A]
Using Strongswan 5.7.0.  The server cert is signed by a Comodo SSL CA.  Android does not include this CA in its OS.  The android client fails now with "no trusted RSA public key for xxxxx".  If I'm correct strongswan verifies the server cert on the client side using the CAs installed in Android.

How can strongswan be forced to recognize the CA or be forced to trust the server cert?

**Comment by Thermi** [comment]
Import the CA file into the Android trust store. You can not disable certificate authentication.

**Comment by dbrody2004** [comment]
Our android strongswan client is a custom app and leverages the frontends/android java example in the distribution.  We cannot ask an user to install a certificate.  However, we modified the code in TrustedCertificateManager.java so that we add the two certificates (CA and intermediate - appears that other code must check the complete the chain, not just the CA) under the loadCertificates() routine.  Tested on an emulator and seems to work.

Would be nice if didn't have to hardcode this the app and that strongswan android had a flag to disable chain checking and just check the CN of the server client.  I know this opens this to MITM, but that would be a risk deployment would take.

**Reply by Thermi** [reply]
Then it wouldn't be ***strong***Swan, would it?
